src/prefect/experimental/workers/base.py

```diff
@@ -0,0 +1,590 @@
+                await deployment.apply()
+                self.logger.info(
+                    "Applied update to deployment %s/%s",
+                    deployment.flow_name,
+                    deployment.name,
+                )
+        except ValidationError:
+            self.logger.exception(
+                "Validation of deployment manifest %s failed.",
+                deployment_manifest_path,
+            )
+        except Exception:
+            self.logger.exception(
+                "Unexpected error occurred while attempting to register discovered deployment."
+            )
+
+    async def get_scheduled_flow_runs(
+        self,
+    ) -> List[schemas.responses.WorkerFlowRunResponse]:
+        """
+        Retrieve scheduled flow runs from the worker pool's queues.
+        """
+        async with get_client() as client:
+            scheduled_before = pendulum.now("utc").add(
+                seconds=int(self.prefetch_seconds)
+            )
+            try:
+                return await client.get_scheduled_flow_runs_for_worker_pool_queues(
+                    worker_pool_name=self.worker_pool_name,
+                    worker_pool_queue_names=[q.name for q in self.worker_pool_queues],
+                    scheduled_before=scheduled_before,
+                )
+            except ObjectNotFound:
+                # the pool doesn't exist; it will be created on the next
+                # heartbeat (or an appropriate warning will be logged)
+                return []
+
+    async def _submit_scheduled_flow_runs(
+        self, flow_run_response: List[WorkerFlowRunResponse]
+    ):
+        """
+        Takes a list of WorkerFlowRunResponses and submits the referenced flow runs
+        for execution by the worker.
+        """
+        for entry in flow_run_response:
+            try:
+                if self.limiter:
+                    self.limiter.acquire_on_behalf_of_nowait(entry.flow_run.id)
+            except anyio.WouldBlock:
+                self.logger.info(
+                    f"Flow run limit reached; {self.limiter.borrowed_tokens} flow runs in progress."
+                )
+                break
+            else:
+                self.logger.info(f"Submitting flow run '{entry.flow_run.id}'")
+                self._submitting_flow_run_ids.add(entry.flow_run.id)
+                self._runs_task_group.start_soon(
+                    self._submit_run,
+                    entry.flow_run,
+                )
+
+    async def _submit_run(self, flow_run: FlowRun) -> None:
+        """
+        Submits a given flow run for execution by the worker.
+        """
+        ready_to_submit = await self._propose_pending_state(flow_run)
+
+        if ready_to_submit:
+            readiness_result = await self._runs_task_group.start(
+                self._submit_run_and_capture_errors, flow_run
+            )
+
+            if readiness_result and not isinstance(readiness_result, Exception):
```

Comment:
What if it's an exception?

Reply:
The exception is handled in `_submit_run_and_capture_errors` where the flow run is either marked as failed if submission failed, or a error is logged saying that something might have gone wrong. This check is to make sure that the worker doesn't try to update the flow run in the backend if something when wrong during submission.

This was taken from the existing agent implementation so this could be a good time to improve this logic if there's a better way to do it.